### PR TITLE
DATAUP-197 app cell cancel after code compile error

### PR DESCRIPTION
**Pull request by eamahanna**
# Description of PR purpose/changes

If a user makes a change to the code in an app cell, and the code does not compile in the app cell, the user can now press cancel. Users were not able to do this before. 

I also added a could of tests to kick off testing for the app cell. This is not comprehensive, and should be added on over time. 

# Jira Ticket / Issue #
https://kbase-jira.atlassian.net/browse/DATAUP-197
- [ ] Added the Jira Ticket to the title of the PR e.g. (DATAUP-69 Adds a PR template)

# Testing Instructions
Execute `make test-frontend-unit`

To test the changes through the ui complete the following steps:
1. open a narrative
2. add an app cell instance to the narrative
3. Open the app cell code, and make a change that will cause the code to fail to compile. 
4. Click run
5. Click cancel when you see the error
6. You can now rerun the cell

- [ ] Tests pass in travis and locally 
- [ ] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [ ] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have run Black and Flake8 on changed Python Code manually or with git precommit (and the travis build passes)

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: nbextensions/appCell2/widgets/appCellWidget-fsm.js [hunk 1]
```diff
@@ -557,7 +557,7 @@ define([], function() {
                 },
                 actionButton: {
                     name: 'cancel',
-                    disabled: true
+                    disabled: false
```
Comment (eamahanna) [inline nbextensions/appCell2/widgets/appCellWidget-fsm.js:560]: Main change to enable the cancel button to be active

File: test/unit/karma.conf.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ module.exports = function (config) {
                 failFast: false,
                 DEFAULT_TIMEOUT_INTERVAL: 20000
             },
-            requireJsShowNoTimestampsError: '^(?!.*(^/narrative/static/))',
+            requireJsShowNoTimestampsError: '^(?!.*(^/narrative/))',
```
Comment (eamahanna) [inline test/unit/karma.conf.js:14]: Edit to see unintended errors when running tests.

**Review by ialarmedalien — APPROVED** [review]
From my very limited knowledge of the narrative UI, this LGTM.

What does '-fsm' mean in the file name?

File: test/unit/karma.conf.js [hunk 3]
```diff
@@ -32,7 +32,7 @@ module.exports = function (config) {
         },
         files: [
             'kbase-extension/static/narrative_paths.js',
-            {pattern: 'test/unit/spec/**/*.js', included: false},
+            {pattern: 'test/unit/spec/nbextensions/appCell2/widgets/*.js', included: false},
```
Comment (bio-boris) [inline test/unit/karma.conf.js:35]: Should this be reverted?

Comment (eamahanna) [reply test/unit/karma.conf.js:35]: YES. my bad. thanks for catching this.

**Comment by eamahanna** [comment]
@ialarmedalien It means "finite state machine". This is what controls the state and state transitions of the app cell.

File: test/unit/karma.conf.js [hunk 4]
```diff
@@ -32,7 +32,7 @@ module.exports = function (config) {
         },
         files: [
             'kbase-extension/static/narrative_paths.js',
-            {pattern: 'test/unit/spec/**/*.js', included: false},
+            {pattern: 'test/unit/spec/**.js', included: false},
```
Comment (bio-boris) [inline test/unit/karma.conf.js:35]: Is this still supposed to be different?

Comment (briehl) [reply test/unit/karma.conf.js:35]: Nope, that's what causing probably only a small handful of tests to run. :)

**Review by briehl — DISMISSED** [review]
Put back the `**/*.js` in `karma.conf.js` line 35, and all should be well.

**Pull request closed without merging**